src/Compilers/CSharp/Portable/FlowAnalysis/NullableWalker.cs

```diff
@@ -403,6 +470,7 @@ private void TrackNullableStateForAssignment(BoundNode node, Symbol assignmentTa
                         {
                             ReportStaticNullCheckingDiagnostics(ErrorCode.WRN_NullReferenceAssignment, (value ?? node).Syntax);
                         }
+                        this.State[slot] = valueIsNotNull;
```

Comment:
>this.State[slot] = valueIsNotNull; [](start = 24, length = 34)

Why is it safe to assume that slot is > 0 here?

Reply:
Perhaps we should simply make the ```else if (slot > 0)``` below into an ```if (slot > 0)```, then we will have consistent and safe handling.

---
In reply to: [152695563](https://github.com/dotnet/roslyn/pull/23063#discussion_r152695563) [](ancestors = 152695563)